### ISSUE TITLE
Timeline, creating range item that spans multiple groups.

### ISSUE THREAD

**Issue by hitmanmcc** [almende/vis#1285]
Hello guys,

First of all, I'd like to thank you for your work on this awesome lib.
Now on to the topic of this issue, which is in fact a feature request/request for information. I'm trying to make a planning component for a hotel application that allows me to display a timeline with rooms (groups) and the reservations assigned to them (item ranges).
I was planning on using the click and drag to enable users to easily create a new reservation for a given room, for a certain period of days. This works fine, if the reservation is just for one room. However, if I want to make a reservation that spans multiple rooms, I cannot because I am unable to find any option or combination of options that would allow me to click and drag to create an item that spans multiple groups in a timeline.

Just so you get a sense of what I'm talking about, I've attached a picture here that shows you how the timeline looks like for reservations that are only for 1 room.
![image](https://cloud.githubusercontent.com/assets/3464842/9764457/33cd251c-5707-11e5-96b3-be55e4a8e598.png)

Best regards,
André

**Comment by josdejong** [comment]
Thanks. RangeItems spanning multiple groups is indeed not supported, we can put this as a feature request.

What you can do is duplicate such a reservation into multiple range items, each in one room, and keep them synced via event listeners.

**Comment by hitmanmcc** [comment]
Hello @josdejong, thanks for your reply. Indeed, that is a possible workaround but I'd have to hack your code a bit. I'm using your library's ctrl+drag functionality to create reservations quickly and easily as that emulates the behaviour we have in our desktop application.
To support the behaviour we want, I'd have to implement a mechanism to assign the number of reservation lines to create before starting the ctrl+drag action and after the action was done, I'd then have to replicate that event across as many groups as we initially specified.
Visually, this would feel strange to the user, to see just one RangeItem being drawn and then a couple of them popping up. Also, having to somehow pre-select the number of rooms to occupy would be kind of cumbersome. It might be a workable solution for some time but, in the future, it would definitely be something for us to revise in order to improve the UX.
I don't know if this is a direction you guys intend your library to take but we'd certainly be very glad if it did. In any case, the simple fact that you were willing to hear us out on this is already a notable fact, so thank you for that. :)

**Comment by josdejong** [comment]
Yes I agree that this looks/works not very smoothly for a user, its really a hack.

RangeItems spanning multiple groups sounds like a good use case to implement in the Timeline, but we can't pick this up ourselves anytime soon.

**Comment by hitmanmcc** [comment]
I'm happy to see you guys label this as an enhancement, regardless of the time frame.

**Comment by yotamberk** [comment]
@hitmanmcc Items spanning over a number of groups means that the item can only span over groups that are beside each-other?  How does one add an item to a group and link it to another group that is not directly near it? 
I feel like this should be implemented by the user like @josdejong mentioned above. I don't think it's a hack or a UX problem if you trigger an event after adding an item and a prompt pops up to ask if you would like to link it to another group (of course this prompt can be popped up to you're specifications and conditions). 
Maybe another way to do it is by having an option to select a number of groups and only then, when dragging (adding) an item, trigger an event to sync an addition of items to the selected groups.

In conclusion, I think the solution to this issue isn't adding the feature for items spanning on multiple groups, but a UX problem that can be implemented in different ways using synchronizations of events on multiple groups (like suggested above)

What do you say?

**Comment by hitmanmcc** [comment]
@yotamberk Hello, we have since abandoned this part of the application and did not pursue this further, I was reading over this to better contextualize my answer. What was intended at the time was for us to be able to perform operations like we do on our desktop PMS software, where we drag-select a square/rectangle like area that covers a group of consecutively numbered rooms and dates.
What this does, in effect, is to create a single reservation spanning multiple rooms, which can later be assigned to the different guests.
If you wish to further discuss this, I'd be happy to answer any question, discuss any idea you put forth. As for your suggestions, although valid, I feel they would be lacking in terms of usability, in our particular scenario. However, I understand it might not be a regular use case for the users of your library, I don't know.

Best regards,
André

**Comment by yotamberk** [comment]
@hitmanmcc Thanks for the response! So let me get this clear:
The feature asked is the ability to multi-select a number of items using a mouse hold-&-drag over a number of groups (kind of like on the Windows desktop selecting a few icons)?

**Comment by hitmanmcc** [comment]
Yes, it is akin to the nature of that process. Sorry for the delay in responding to your query.
